plasmapy/physics/dielectric.py

```diff
@@ -0,0 +1,182 @@
+"""Functions to calculate plasma dielectric paramaters"""
+
+from astropy import units
+import plasmapy.utils as utils
+
+from ..constants import (pi, m_e, c, mu0, e, eps0)
+from .parameters import gyrofrequency, plasma_frequency
+
+r"""
+Values should be returned as an Astropy Quantity in SI units.
+"""
+
+
+@utils.check_quantity({
+    'B': {'units': units.T, 'can_be_negative': False},
+    'omega': {'units': units.rad/units.s, 'can_be_negative': False},
+})
+def cold_plasma_permittivity_SDP(B, species, n, omega):
+    """
+    Magnetized Cold Plasma Dielectric Permittivity Tensor Elements.
+    Elements (S, D, P) are given in the "Stix" frame, ie. with B // z.
+
+    The :math:`\exp(-i \omega t)` time-harmonic convention is assumed.
+
+    References
+    - T.H. Stix, Waves in Plasma, 1992.
+
+    Parameters
+    -------
+    B : ~astropy.units.Quantity
+        Magnetic field magnitude in units convertible to tesla.
+
+    species : List of string
+        List of the plasma particle species
+        e.g.: ['e', 'D+'] or ['e', 'D+', 'He+'].
+
+    n : List of ~astropy.units.Quantity
+        List of species density in units convertible to per cubic meter
+        The order of the species densities should follow species.
+
+    omega : ~astropy.units.Quantity
+        Electromagnetic wave frequency in rad/s
+
+    Returns
+    -------
+    S : ~astropy.units.Quantity
+        S ("Sum") dielectric tensor element
+
+    D : ~astropy.units.Quantity
+        D ("Difference") dielectric tensor element
+
+    P : ~astropy.units.Quantity
+        P ("Plasma") dielectric tensor element
+
+    Notes
+    -----
+    The dielectric permittivity tensor is expressed in the Stix frame with
+    the :math:`\exp(-i \omega t)` time-harmonic convention as:
+
+    .. math::
+        :nowrap:
+
+        \varepsilon =
+        \varepsilon_0
+        \left(
+        \begin{array}{ccc}
+        S & -i D & 0 \\
+        +i D & S & 0 \\
+        0 & 0 & P
+        \end{array}
+        \right)
+
+    with:
+
+    .. math::
+        S = 1 - \sum_s \frac{\omega_{p,s}^2}{\omega^2 - \Omega_{c,s}^2}
+
+        D = \sum_s \frac{\Omega_{c,s}}{\omega} \frac{\omega_{p,s}^2}{\omega^2 - \Omega_{c,s}^2}
+
+        P = 1 - \sum_s \frac{\omega_{p,s}^2}{\omega^2}
+
+    where :math:`\omega_{p,s}` is the plasma frequency and
+    :math:`\Omega_{c,s}` is the signed version of the cyclotron frequency
+    for the species :math:`s`.
+
+    Examples
+    --------
+    >>> B = 2*u.T
+    >>> species = ['e', 'D+']
+    >>> n = [1e18*u.m**-3, 1e18*u.m**-3]
+    >>> omega = 50e6*(2*np.pi)*(u.rad/u.s)
+    >>> S, D, P = cold_plasma_permittivity_SDP(B, species, n, omega)
+    ￼
+    """
+    S, D, P = 1, 0, 1
+
+    for (idx, s) in enumerate(species):
+        omega_c = gyrofrequency(B=B, particle=s, signed=True)
+        omega_p = plasma_frequency(n=n[idx], particle=s)
+
+        S += - omega_p**2 / (omega**2 - omega_c**2)
+        D += omega_c/omega * omega_p**2 / (omega**2 - omega_c**2)
```

Comment:
Please add spaces between operators and variables e.g., `omega_p ** 2` instead of `omega_p**2`.